torch/csrc/tensor/python_tensor.cpp

```diff
@@ -277,6 +277,11 @@ static THPObjectPtr get_tensor_dict() {
 static std::vector<PyTensorType*> tensor_types;
 
 void set_default_tensor_type(PyTensorType* type) {
+  // Make sure this is not a null pointer before accessing it.
+  if (!type) {
+    throw TypeError("invalid tensor type");
```

Comment:
"Unsupported default tensor dtype"?

Reply:
nah, i don't think this is better. because unsupported somehow implies that the pointer is valid and the type is not valid. and my version stresses out that we just have nothing to work with, for whatever reason.

Reply:
i will switch to "invalid default tensor dtype", i think this combines the pros of both variants